### DISCUSSION TITLE
How to access array-indices with json_parser operator?

### DISCUSSION THREAD

**Discussion by FWinkler79** [open-telemetry/opentelemetry-collector-contrib#26121] [Q&A]
Hi there,

I am struggling to find out how to access json objects that are within an array using json_parser.
What I have is a structure like this (in the attributes field):

```json
"agents": [
        {
            "active": 1,
            "metricsAtStart": "Error:  This is an error message",
            "metricsAtEnd": "",
            "agentId": 1089681,
            "agentName": "some name",
            "dateStart": "2023-08-28 14:52:00",
        }
    ]
```
... and there may potentially be many such object.

Now with json_parser I was trying to do the following:

```yaml
operators:
      - type: json_parser
        parse_from: body    
        body: attributes.agents[0]
```

But this will fail at startup of the collector, with the following error:
```
* error decoding 'from': splitting field: strings in brackets must be surrounded by quotes
```

When I use quotes to embrace the array index (0) the collector will start but not extract the first object in the array.
What's the proper syntax to access array indices?

Thanks!

**Comment by nielsout** [comment]
If you know how many items the array has, you can do:
```
      - type: assign_keys
        field: "attributes.agents"
        id: parse_agents
        keys: ["foo"] # amount of keys has to equal the number of items in the array
        if: attributes.agents != nil
        output: parse_foo
```
After this, attributes.agents.foo gives you access to the objects. 

I'm running into the same issue with an array with a varying number of items, has anyone found a solution for that?
